autoSyncWorker: null pointer exception causing looper problem and toasting error

Work [ id=6e3fd0e0-9a35-4a17-ac01-8c08de3c6671, tags={ org.ole.planet.myplanet.service.AutoSyncWorker } ] failed because it threw an exception/error                                                                                                    java.util.concurrent.ExecutionException: java.lang.NullPointerException: Can't toast on a thread that has not called Looper.prepare()
at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:516)
at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:475)
at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:317)
at androidx.work.impl.utils.SerialExecutorImpl$Task.run(SerialExecutorImpl.java:96)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.NullPointerException: Can't toast on a thread that has not called Looper.prepare()
at com.android.internal.util.Preconditions.checkNotNull(Preconditions.java:167)
at android.widget.Toast.getLooper(Toast.java:265)
at android.widget.Toast.<init>(Toast.java:250)
at android.widget.Toast.makeText(Toast.java:756)
at android.widget.Toast.makeText(Toast.java:744)
at org.ole.planet.myplanet.utilities.Utilities.toast(Utilities.java:82)
at org.ole.planet.myplanet.service.AutoSyncWorker.doWork(AutoSyncWorker.java:54)
at androidx.work.Worker$1.run(Worker.java:82)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137) 
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637) 
at java.lang.Thread.run(Thread.java:1012) 

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -10,8 +10,8 @@ android {
         applicationId "org.ole.planet.myplanet"
         minSdkVersion 19
         targetSdkVersion 33
-        versionCode 971
-        versionName "0.9.71"
+        versionCode 985
+        versionName "0.9.85"
         ndkVersion '21.3.6528147'
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true

--- a/app/src/main/java/org/ole/planet/myplanet/service/AutoSyncWorker.java
+++ b/app/src/main/java/org/ole/planet/myplanet/service/AutoSyncWorker.java
@@ -9,6 +9,8 @@ import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
 import android.content.SharedPreferences;
+import android.os.Handler;
+import android.os.Looper;
 
 import androidx.annotation.NonNull;
 import androidx.work.Worker;
@@ -46,10 +48,13 @@ public class AutoSyncWorker extends Worker implements SyncListener, Service.Chec
         preferences = context.getSharedPreferences(SyncManager.PREFS_NAME, MODE_PRIVATE);
         long lastSync = preferences.getLong("LastSync", 0);
         long currentTime = System.currentTimeMillis();
-        int syncInterval = preferences.getInt("autoSyncInterval", 60 * 60);
 
-        if ((currentTime - lastSync) > (syncInterval * 1000)) {
-            Utilities.toast(context, "Syncing started...");
+        if ((currentTime - lastSync) > (30)) {
+            // Post a Runnable to the main thread's Handler to show the Toast
+            Handler mainHandler = new Handler(Looper.getMainLooper());
+            mainHandler.post(() -> {
+                Utilities.toast(context, "Syncing started...");
+            });
             new Service(context).checkVersion(this, preferences);
         }
 

--- a/app/src/main/java/org/ole/planet/myplanet/service/AutoSyncWorker.java
+++ b/app/src/main/java/org/ole/planet/myplanet/service/AutoSyncWorker.java
@@ -48,8 +48,9 @@ public class AutoSyncWorker extends Worker implements SyncListener, Service.Chec
         preferences = context.getSharedPreferences(SyncManager.PREFS_NAME, MODE_PRIVATE);
         long lastSync = preferences.getLong("LastSync", 0);
         long currentTime = System.currentTimeMillis();
+        int syncInterval = preferences.getInt("autoSyncInterval", 60 * 60);
 
-        if ((currentTime - lastSync) > (30)) {
+        if ((currentTime - lastSync) > (syncInterval * 1000)) {
             // Post a Runnable to the main thread's Handler to show the Toast
             Handler mainHandler = new Handler(Looper.getMainLooper());
             mainHandler.post(() -> {

--- a/app/src/main/res/values/versions.xml
+++ b/app/src/main/res/values/versions.xml
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="app_version">0.9.71</string>
+    <string name="app_version">0.9.85</string>
 </resources>

